lib/mayaUsd/fileio/utils/meshReadUtils.cpp

```diff
@@ -888,17 +894,66 @@ MStatus UsdMayaMeshReadUtils::assignSubDivTagsToMesh(
 
 #if MAYA_API_VERSION >= 20220000
 
-MStatus
-UsdMayaMeshReadUtils::getComponentTags(const UsdGeomMesh& mesh, std::vector<ComponentTagData>& tags)
+bool UsdMayaMeshReadUtils::getGeomSubsetInfo(const MObject& mesh, JsValue& meshRoundtripData)
+{
+    MFnDependencyNode depNodeFn(mesh);
+    MPlug             plug = depNodeFn.findPlug(MString(_meshTokens->USD_GeomSubsetInfo.GetText()));
+    if (!plug.isNull()) {
+        JsParseError error;
+        JsValue      value = JsParseString(plug.asString().asChar(), &error);
+        if (value) {
+            meshRoundtripData = value;
+            return true;
+        }
+    }
+
+    return false;
+}
+
+void UsdMayaMeshReadUtils::setGeomSubsetInfo(const MObject& mesh, const JsValue& meshRoundtripData)
+{
+    MStatus           status { MS::kSuccess };
+    MFnDependencyNode depNodeFn(mesh);
+    MPlug             plug = depNodeFn.findPlug(MString(_meshTokens->USD_GeomSubsetInfo.GetText()));
+    if (plug.isNull()) {
+
+        MFnTypedAttribute tAttr;
+        MObject           attr = tAttr.create(
+            _meshTokens->USD_GeomSubsetInfo.GetText(),
+            _meshTokens->USD_GeomSubsetInfo.GetText(),
+            MFnData::kString,
+            MObject::kNullObj,
+            &status);
+        if (status == MS::kSuccess) {
+            depNodeFn.addAttribute(attr);
+            plug = depNodeFn.findPlug(attr);
+        }
+    }
+    if (!plug.isNull()) {
+        std::ostringstream ostr;
+        JsWriter           compactWriter(ostr, JsWriter::Style::Compact);
+        JsWriteValue(&compactWriter, meshRoundtripData);
+        plug.setString(ostr.str().c_str());
+    }
+}
+
+MStatus UsdMayaMeshReadUtils::getComponentTags(
+    const UsdGeomMesh&             mesh,
+    std::vector<ComponentTagData>& tags,
+    JsValue&                       meshRoundtripData)
 {
     MStatus status { MS::kSuccess };
 
     // Find all the prims defining componentTags
-    TfToken                    componentTagFamilyName("componentTag");
-    std::vector<UsdGeomSubset> subsets
-        = UsdGeomSubset::GetGeomSubsets(mesh, UsdGeomTokens->face, componentTagFamilyName);
+    std::vector<UsdGeomSubset> subsets = UsdGeomSubset::GetGeomSubsets(mesh, UsdGeomTokens->face);
+
+    JsObject allRoundtripData;
 
     for (auto& ss : subsets) {
+        if (UsdMayaRoundTripUtil::IsPrimMayaGenerated(ss.GetPrim())) {
```

Comment:
Wish there were a comment explaining why we're skipping these. Only add if you need another code version due to other things.